Userland/Libraries/LibCore/FileLikeIODevice.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, Ali Mohammad Pur <ali.mpfard@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/EnumBits.h>
+#include <AK/Forward.h>
+#include <LibCore/IODevice.h>
+#include <LibCore/Notifier.h>
+#include <LibCore/Object.h>
+#include <errno.h>
+
+namespace Core {
+
+enum class OpenMode : unsigned {
+    NotOpen = 0,
+    ReadOnly = 1,
+    WriteOnly = 2,
+    ReadWrite = 3,
+    Append = 4,
+    Truncate = 8,
+    MustBeNew = 16,
+};
+
+enum class SeekMode {
+    SetPosition,
+    FromCurrentPosition,
+    FromEndPosition,
+};
+
+AK_ENUM_BITWISE_OPERATORS(OpenMode)
+
+class FileLikeIODevice : public virtual IODevice {
+    C_OBJECT_ABSTRACT(FileLikeIODevice)
+public:
+    // ^IODevice
+    size_t read(Bytes) override;
+    bool discard_or_error(size_t count) override;
+    bool unreliable_eof() const override { return m_eof; }
+    size_t write(ReadonlyBytes) override;
+    RefPtr<AbstractNotifier> make_notifier(unsigned event_mask) override;
```

Comment:
Missing `virtual` ([coding style](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#overriding-virtual-methods))

Reply:
Blame clion, it annoyed me with "hurr durr `virtual` is redundant"